src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -566,49 +575,138 @@ LRESULT CALLBACK FancyZones::s_WndProc(HWND window, UINT message, WPARAM wparam,
+    if(GetSystemMetrics(SM_REMOTESESSION))
+    {
+        // When running in a remote session, devices have "Default_Monitor#<transient_part>" DeviceIDS.
+        // Since a remote session can have multiple devices, we can't just assign them a single "REMOTE" ID.
+        // We'll assign them unique numeric IDs instead.
+        for (size_t i = 0; i < size(devices); ++i)
+        {
+            wchar_t remoteDeviceID[32];
+            swprintf_s(remoteDeviceID, L"REMOTE%zu", i);
+            AddZoneWindow(devices[i]._handle, remoteDeviceID);
+        }
+        return;
+    }
+    // If EnumDisplayMonitors returned a duplicate data, it's an API bug and we should workaround it
+    const bool enumDisplayMonitorsDublicateData = [&]{
+        std::unordered_set<std::wstring_view> discoveredDeviceIDs;
+        for (const auto & dev : devices)
+        {
+            const auto generatedPartOfID = dev.get_generated_part_of_id();
+            if (!generatedPartOfID.has_value())
             {
-                if (!deviceId)
-                {
-                    deviceId = GetSystemMetrics(SM_REMOTESESSION) ?
-                        L"\\\\?\\DISPLAY#REMOTEDISPLAY#" :
-                        L"\\\\?\\DISPLAY#LOCALDISPLAY#";
-                }
-
-                auto strongThis = reinterpret_cast<FancyZones*>(data);
-                strongThis->AddZoneWindow(monitor, deviceId);
+                continue;
+            }
+            const auto [_, hasUniqueID] = discoveredDeviceIDs.emplace(*generatedPartOfID);
+            if (!hasUniqueID)
+            {
+                return true;
             }
         }
-        return TRUE;
-    };
+        return false;
+    }();
 
+
+    for (size_t i = 0; i < size(devices); ++i)
     {
-        std::unique_lock writeLock(m_lock);
-        m_zoneWindowMap.clear();
+        const auto & dev = devices[i];
+        const auto generatedPartOfID = dev.get_generated_part_of_id();
+        // In case DeviceID doesn't contain valid data or we can't rely on it, since EnumDisplayMonitors
+        // returned a duplicate results, we try using WMI data or fallback to UNKNOWN + numeric ID.
+        if (!generatedPartOfID.has_value() || enumDisplayMonitorsDublicateData)
+        {
+            const bool hasWmiData = i < size(monitor_infos);
+            if (hasWmiData)
+            {
+                AddZoneWindow(dev._handle, monitor_infos[i].hardware_id().c_str());
+            }
+            else
+            {
+                // no WMI data - likely we're running inside a VM
+                wchar_t localDeviceID[32];
+                swprintf_s(localDeviceID, L"UNKNOWN%zu", i);
+                AddZoneWindow(dev._handle, localDeviceID);
+            }
+            continue;
+        }
+        // The happy path is to associate generated parts of DeviceIDs with a WMIMonitorID.InstanceName
+        // And use WMI IDs instead. We can't get WMI data if we're running inside a VM, but at this point we've

```

Comment:
Can you rephrase `The happy path is to associate generated parts of DeviceIDs with a WMIMonitorID.InstanceNameAnd use WMI IDs instead.`?
Make it as long as needed to explain each step of the process.